src/lib/storage/table.cpp

```diff
@@ -208,14 +208,20 @@ void Table::append_chunk(const Segments& segments, std::shared_ptr<MvccData> mvc
   Assert(_type != TableType::Data || static_cast<bool>(mvcc_data) == (_use_mvcc == UseMvcc::Yes),
          "Supply MvccData to data Tables iff MVCC is enabled");
 
-#if HYRISE_DEBUG
-  for (const auto& segment : segments) {
-    const auto is_reference_segment = std::dynamic_pointer_cast<ReferenceSegment>(segment) != nullptr;
-    DebugAssert(is_reference_segment == (_type == TableType::References), "Invalid Segment type");
+  if constexpr (HYRISE_DEBUG) {
+    for (const auto& segment : segments) {
+      const auto is_reference_segment = std::dynamic_pointer_cast<ReferenceSegment>(segment) != nullptr;
+      Assert(is_reference_segment == (_type == TableType::References), "Invalid Segment type");
+    }
```

Comment:
I suspect that the constexpr changes do not really relate to the rest of this PR but am OK of sneaking them in.

Reply:
Nope, but changing an `#if` to a `if constexpr` hardly warrants its own PR. People may argue "but what if we need to revert this commit", but in the history of this repository, there has been only a single revert on master.